### ISSUE TITLE
Copy/Move/Swap use pre-selected button if one is already selected

### ISSUE THREAD

**Issue by peternewman** [bitfocus/companion#2974]
### Is this a feature relevant to companion itself, and not a module?

- [X] I believe this to be a feature for companion, not a module

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the feature

Currently in your favourite text editor, you select some text, then copy it.
Currently in Companion you select a button whilst doing something, if you press copy it then deselects it and then you have to reselect it to copy it, then click where you want it to go.
If a pre-selected button was used as the first input to Copy/Move/Swap it would be more consistent with other tools and fewer button presses.

### Usecases

Reduce clicks for simple tasks

Potentially not for delete as it's a higher risk

**Comment by Julusian** [comment]
If we are going to do this, it should be explicit and require always selecting the button before the operation.

This is already possible with the keyboard shortcuts

**Comment by dnmeid** [comment]
The workflow of "select action" -> "select source" -> "select target" has been chosen deliberately. It is the workflow used by lighting consoles and we had been orienting at grandMA for a lot of things. So it is consistent with the tools of the event industry.
As Julian says, the copy/paste keyboard shortcuts are working exactly like you are expecting. So for me that is a won't fix.

**Comment by gbdlin** [comment]
Alternatively, copy, move and swap buttons can also be added next to test and delete buttons on right panel. Delete is already duplicated this way and it's pretty obvious that this will trigger the action for the currently selected button.

**Comment by stevekmcc** [comment]
I agree that this kind of decision should be made based on what users are already familiar with. But how many new users are coming from grandMA, and how many from normal computer or mobile applications - where "select object" -> "select action" has been the norm for [decades](https://uxmastery.com/object-focused-vs-task-focused/#:~:text=Pioneers%20in%20GUI%20design%20in%20the%201980s,Hard%20to%20Learn%20%5BPDF%5D%20by%20Dedre%20Gentner).), because it's proven to be more intuitive. I'd suggest this could be an option, with the default for new installations eventually being to allow selecting the object first. 

If an object is selected, take that as the primary focus of the action - whether the action is triggered by a keyboard shortcut or a UI button. For actions requiring a second object, request that. For actions requiring confirmation, like delete, confirm. Apply action.
